frontend/src/metabase/dashboard/actions/getNewCardUrl.ts

```diff
@@ -42,26 +42,45 @@ export const getNewCardUrl = ({
 
   const previousQuestion = new Question(previousCard, metadata);
   const { isEditable } = Lib.queryDisplayInfo(previousQuestion.query());
-  const nextQuestion = isEditable
-    ? new Question(cardAfterClick, metadata)
-        .setDisplay(cardAfterClick.display || previousCard.display)
-        .setSettings(dashcard.card.visualization_settings)
-        .lockDisplay()
-    : new Question(dashcard.card, metadata).setDashboardProps({
-        dashboardId: dashboard.id,
-        dashcardId: dashcard.id,
-      });
-
   const parametersMappedToCard = getParametersMappedToDashcard(
     dashboard.parameters,
     dashcard,
   );
 
+  let nextQuestion: Question | undefined = undefined;
+
+  if (isEditable) {
+    nextQuestion = new Question(cardAfterClick, metadata);
+
+    const isDashcardTitleClick =
+      nextCard.dataset_query === previousCard.dataset_query;
```

Comment:
This change may be controversial even though it works.

Dashcard title click drill is the only drill that does not change the query (at this stage).
Parameters are applied later, when generating URL in `getNewCardUrl`.

Reply:
There is Lib.areLegacyQueriesEqual for this

Reply:
Btw, why do we need it here? I thought that when parameters are applied, we handle extra stages

Reply:
I don't have a satisfying answer now.
Without it few tests were failing - see [CI run](https://github.com/metabase/metabase/actions/runs/11270092393) - mostly click behavior & default display type.
I'll try to come up with a better answer next week.

_I haven't gotten into testing click behavior yet, so this code may change if more issues get discovered._

Reply:
> Btw, why do we need it here?

We don't need it after all: #48726

> Without it few tests were failing

This was because of #48787